metrics/web/lib/project_groups/domain/entities/project_group.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:meta/meta.dart';
+
+/// Represents the project group entity.
+@immutable
+class ProjectGroup {
+  final String id;
+  final String name;
+  final List<String> projectIds;
+
+  /// Creates the [ProjectGroup]
+  ///
+  /// [id] is the unique identifier of this project group.
+  /// [name] is the name of this project group.
+  /// [projectIds] is the list of project ids related with this project group.
+  const ProjectGroup({
+    this.id,
+    @required this.name,
+    @required this.projectIds,
+  })  : assert(name != null),
+        assert(projectIds != null);
```

Comment:
It seems like it would be better to use `ArgumentError` instead of asserts for the domain layer.